### PR TITLE
support ethash mining with --test

### PR DESCRIPTION
**Pull request by winsvega**

### CODE REVIEW

File: libethereum/ClientTest.cpp [hunk 1]
```diff
@@ -58,8 +58,9 @@ void ClientTest::setChainParams(string const& _genesis)
     try
     {
         params = params.loadConfig(_genesis);
-        if (params.sealEngineName != "NoProof")
-            BOOST_THROW_EXCEPTION(ChainParamsNotNoProof() << errinfo_comment("Provided configuration is not well formatted."));
+        if (params.sealEngineName != "NoProof" && params.sealEngineName != "Ethash")
+            BOOST_THROW_EXCEPTION(ChainParamsInvalid() << errinfo_comment(
+                                      "Provided configuration is not well formatted."));
```
Comment (chfast) [inline libethereum/ClientTest.cpp]: This error does not explains what's wrong.

**Comment by gumb0** [comment]
NoProof is still default if I just run `aleth --test`, isnt't it?

File: libethereum/ClientTest.cpp [hunk 2]
```diff
@@ -58,8 +58,9 @@ void ClientTest::setChainParams(string const& _genesis)
     try
     {
         params = params.loadConfig(_genesis);
-        if (params.sealEngineName != "NoProof")
-            BOOST_THROW_EXCEPTION(ChainParamsNotNoProof() << errinfo_comment("Provided configuration is not well formatted."));
+        if (params.sealEngineName != "NoProof" && params.sealEngineName != "Ethash")
+            BOOST_THROW_EXCEPTION(
+                ChainParamsInvalid() << errinfo_comment("Seal engine is not supported!"));
```
Comment (gumb0) [inline libethereum/ClientTest.cpp]: Remove `ChainParamsNotNoProof` definition if it's not used nowhere else

**Comment by gumb0** [comment]
I'm working on fixing the test failure in rpc tests

**Pull request merged**